src/API/Management/ResourceServers.php

```diff
@@ -2,72 +2,150 @@
+ *
+ * @package Auth0\SDK\API\Management
+ */
 class ResourceServers extends GenericResource
 {
     /**
+     * Get all Resource Servers, by page if desired.
+     * Required scope: "read:resource_servers"
+     *
+     * @param null|integer $page     Page number to get, zero-based.
+     * @param null|integer $per_page Number of results to get, null to return the default number.
      *
      * @return mixed
+     *
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Resource_Servers/get_resource_servers
      */
-    public function getAll()
+    public function getAll($page = null, $per_page = null)
     {
-        return $this->apiClient->get()
-        ->addPath('resource-servers')
-        ->call();
+        $params = [];
+
+        // Pagination parameters.
+        if (null !== $page) {
+            $params['page'] = abs(intval($page));
+        }
+
+        if (null !== $per_page) {
+            $params['per_page'] = abs(intval($per_page));
+        }
+
+        return $this->apiClient->method('get')
+            ->withDictParams($params)
+            ->addPath('resource-servers')
+            ->call();
     }
 
     /**
+     * Get a single Resource Server by ID or API identifier.
+     * Required scope: "read:resource_servers"
+     *
+     * @param string $id Resource Server ID or identifier to get.
      *
-     * @param  string $id
      * @return mixed
+     *
+     * @throws CoreException Thrown if the id parameter is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Resource_Servers/get_resource_servers_by_id
      */
     public function get($id)
     {
-        return $this->apiClient->get()
-        ->addPath('resource-servers', $id)
-        ->call();
+        if (empty($id) || ! is_string($id)) {
+            throw new CoreException('Invalid "id" parameter.');
+        }
+
+        return $this->apiClient->method('get')
+            ->addPath('resource-servers', $id)
+            ->call();
     }
 
     /**
+     * Create a new Resource Server.
+     * Required scope: "create:resource_servers"
+     *
+     * @param string $identifier API identifier to use.
+     * @param array  $data       Additional fields to add.
      *
-     * @param  string $client_id
-     * @param  array  $data
      * @return mixed
+     *
+     * @throws CoreException Thrown if the identifier parameter or data field is empty or is not a string.
+     * @throws \Exception Thrown by the HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Resource_Servers/post_resource_servers
      */
-    public function create($client_id, $data)
+    public function create($identifier, array $data)
```

Comment:
Better param name, not breaking

Reply:
$client_id was never used before ? 😆 Backwards compatibility seems fine.

Reply:
Yeah, param was there but not used anywhere in the method 🤷‍♂️